src/test/java/org/mapdb/io/DataIOTest.java

```diff
@@ -335,4 +366,295 @@ public void testUnpackLong_withInputStream_throws_exception_when_stream_is_empty
                 DataIO.packLongSize(1 << 63));
     }
 
-}
\ No newline at end of file
+    @Test public void testUnpackInt() throws IOException {
+        Assert.assertEquals(126, DataIO.unpackInt(
+                new ByteArrayInputStream(new byte[]{-2, -15, -128})));
+        Assert.assertEquals(0, DataIO.unpackInt((DataInput) new
+                DataInputStream(new ByteArrayInputStream(new byte[]{-128}))));
+
+        Assert.assertEquals(0, DataIO.unpackInt(new byte[]{-128}, 0));
+        Assert.assertEquals(0, DataIO.unpackInt(new byte[]{0, -128}, 0));
+    }
+
+    @Test public void testUnpackIntThrowsException1() throws IOException {
+        thrown.expect(EOFException.class);
+        DataIO.unpackInt((DataInput)new DataInputStream(
+                new ByteArrayInputStream(new byte[]{0})));
+    }
+
+    @Test public void testUnpackIntThrowsException2() throws IOException {
+        thrown.expect(EOFException.class);
+        DataIO.unpackInt(new ByteArrayInputStream(new byte[]{0}));
+    }
+
+    @Test public void testUnpackLong() throws IOException {
+        Assert.assertEquals(0L, DataIO.unpackLong(
+                new ByteArrayInputStream(new byte[]{0, -128, -127})));
+        Assert.assertEquals(0L, DataIO.unpackLong(
+                new ByteArrayInputStream(new byte[]{-128, -127})));
+        Assert.assertEquals(0, DataIO.unpackLong((DataInput) new
+                DataInputStream(new ByteArrayInputStream(new byte[]{-128}))));
+
+        Assert.assertEquals(0L, DataIO.unpackLong(new byte[]{-128}, 0));
+        Assert.assertEquals(0L, DataIO.unpackLong(new byte[]{0, -128}, 0));
+    }
+
+    @Test public void testUnpackLongThrowsException() throws IOException {
+        thrown.expect(EOFException.class);
+        DataIO.unpackLong(new ByteArrayInputStream(new byte[]{0}));
+    }
+
+    @Test public void testPackLong() {
+        Assert.assertArrayEquals(new byte[]{(byte)0, (byte)0},
```

Comment:
These casts `(byte)0` are not needed. If the array is full of zeroes, it can be instantiated with `new byte(_length_)` where _length_ is the number of elements in the array.
Similarly for other arrays used in this test class.

Reply:
changed